File: swift/Sources/sdnx/Types/FieldSchema.swift
import Foundation

public struct FieldSchema: SchemaValue {
    public let type: String
    public var description: String?
    public var validators: [String: ValidatorInfo]?
    
    public init(type: String, description: String? = nil, validators: [String: ValidatorInfo]? = nil) {
        self.type = type
        self.description = description
        self.validators = validators
    }
}

File: zig/src/types/FieldSchema.zig
const std = @import("std");
const Validator = @import("./Validator.zig").Validator;

/// Field schema represents a single field with type and validators
pub const FieldSchema = struct {
    type: []const u8,
    description: ?[]const u8 = null,
    validators: ?std.StringArrayHashMap(Validator) = null,
};
